caffe2/CMakeLists.txt

```diff
@@ -1310,6 +1330,14 @@ if(BUILD_MOBILE_TEST)
   endforeach()
 endif()
 
+add_executable(lite_interpreter_demo ${TORCH_ROOT}/test/cpp/demos/main.cpp)
+target_link_libraries(lite_interpreter_demo torch_library)
```

Comment:
where is `torch_library` defined?

Reply:
It's defined here (https://github.com/pytorch/pytorch/blob/4bbddb45e37c389a911a421989df757257b650b8/caffe2/CMakeLists.txt#L1173).  `torch_library` is an interface library that indirectly depends on the real library `torch`, and `torch` is the `libtorch` library target. With this change, if the flag `BUILD_LITE_INTERPRETER ` is on,
the libtorch library will have the same name, but the source code will be lite interpreter. 

It's probably better to rename it to `libtorch_lite`, but I have some concern regarding the dependency, so I keep it as is in this pr.